src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Here the Delay Timer is stopped in `Mle::Stop()` which force the device to be detached, I think it's not a normal detach, we can't control the time period from next 'thread start' and how many network changes may happen during the period, so I think it is not necessary to start mLocal's Delay Timer here, just try to store current datasets and wait for next `thread start`;

And for Certification test, I don't want the Active Dataset be overwritten by Pending Dataset after 'thread stop', so we could have both Active Dataset and Pending Dataset stored before next 'thread start' and start the reattach process with them.

At the same time, although the Delay Timer is stopped here, we still can make sure the Pending Dataset will take effect in all cases after "thread start":
- If device re-sync to network with `Child Synchronization`/`Link Synchronization` process successfully, Delay Timer will be restarted in `SetStateChild()`/`SetStateRouter()`;
- If device fails to re-sync, it will start reattach process with Active Dataset firstly, if successfully attach, Delay Timer will be restarted in `SetStateChild()`/`SetStateRouter()`;
- If device fails to attach using Active Dataset, it will try to attach using Pending Dataset, so Pending Dataset takes effect here;
- If device fails to attach using Pending Dataset, it will revert back to using Active Dataset and start Delay Timer for Pending Dataset, Pending Dataset will take effect finally.

Please correct me if I missed something, thanks a lot for your time and patient.